cmake/cares.cmake

```diff
@@ -26,10 +26,16 @@ if(gRPC_CARES_PROVIDER STREQUAL "module")
 
   if(TARGET c-ares)
     set(_gRPC_CARES_LIBRARIES c-ares)
+    if(NOT CMAKE_VERSION VERSION_LESS 3.13)
```

Comment:
I don't like that we're comparing CMAKE_VERSION to 3.13 in ten places - that's very hard to maintain.
I suggest adding a variable `_gRPC_INSTALL_SUPPORTED_FROM_MODULE` in the CMakeLists.txt.template  and use it in all all the `.cmake` files, otherwise the logic is spread across too many places.